src/did/did_doc.rs

```diff
@@ -0,0 +1,63 @@
+//! Set of interfaces that describe DID Document (https://www.w3.org/TR/did-core/)
+
+use serde_json::Value;
+
+/// Represents DID Document (https://www.w3.org/TR/did-core/)
+pub trait DIDDoc {
+    /// Returns DID URLs of verification methods used for key agreement.
+    /// See https://www.w3.org/TR/did-core/#verification-methods.
+    fn key_agreements(&self) -> &[String];
+
+    /// Returns DID URLs of verification methods used for authentication.
+    /// See https://www.w3.org/TR/did-core/#authentication
+    fn authentications(&self) -> &[String];
+
+    /// Returns all local verification methods including embedded to
+    /// key agreement and authentication sections.
+    /// See https://www.w3.org/TR/did-core/#verification-methods.
+    fn verification_methods(&self) -> &[VerificationMethod];
```

Comment:
I prefer `verification_method(kid)` because of the following reasons:
- verification_methods forces the user to handle and parse ALL verification methods in a DID Doc, even ones that we don't need (neither authentication nor keyAgreement)
- from the usage point of view verification_method can be even easier. For example in Python it will be something like this:
  `key_agreements = [verification_method[kid] for kid in key_agreements()]`
  `key_agreement = verification_method[kid]`

Reply:
After talking decided to leave as is.